libpod/options.go

```diff
@@ -636,3 +637,58 @@ func WithPodLabels(labels map[string]string) PodCreateOption {
 		return nil
 	}
 }
+
+// WithDNSSearch sets the additional search domains of a container
+func WithDNSSearch(searchDomains []string) CtrCreateOption {
+	return func(ctr *Container) error {
+		if ctr.valid {
+			return ErrCtrFinalized
+		}
+		if StringInSlice(".", searchDomains) && len(searchDomains) > 1 {
+			return errors.Errorf("cannot pass additional search domains when also specifying '.'")
+		}
+		ctr.config.DNSSearch = searchDomains
+		return nil
+	}
+}
+
+// WithDNS sets additional name servers for the container
+func WithDNS(dnsServers []string) CtrCreateOption {
+	return func(ctr *Container) error {
+		if ctr.valid {
+			return ErrCtrFinalized
+		}
+		var dns []net.IP
+		for _, i := range dnsServers {
```

Comment:
Should we check to make sure they gave us at least 1 DNS server here? Are there cases where having no DNS servers is something you want?

Reply:
this only gets called if they pass at least one in the first place